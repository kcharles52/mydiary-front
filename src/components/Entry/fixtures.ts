export const props = {
  entry_id: 1,
  title: "Title 1",
  diaryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete:jest.fn()

};
