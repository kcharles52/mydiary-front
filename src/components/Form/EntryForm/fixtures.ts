export const EntryFormProps = {
  Edit: false,
  title: "string",
  diaryTitle: "string",
  date: "string",
  diaryBody: "string",
  onSubmit: jest.fn(),
  onCancel: jest.fn(),
  onChange: jest.fn()
};

  