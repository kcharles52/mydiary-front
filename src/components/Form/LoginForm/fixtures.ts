export const LoginFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  