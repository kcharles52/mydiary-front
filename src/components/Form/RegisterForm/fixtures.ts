export const RegisterFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  