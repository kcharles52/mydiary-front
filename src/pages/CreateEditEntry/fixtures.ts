export const props = {
  Edit: false,
  title: "Form title",
  diaryTitle: "Test Diary Title",
  date: "20-2-19",
  diaryBody: "Test Diary body",
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  addEntryAction: jest.fn(),
  editEntryAction: jest.fn(),
  getEntryAction: jest.fn(() =>
    Promise.resolve({
      entry: {
        entry_id: 1,
        diaryTitle: "Title 1",
        diaryEntryBody:
          "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
        date: "12/12/2018",
        delete: jest.fn()
      }
    })
  ),
  match: { params: { entry_id: "1" } },
  editEntry: {
    entry: {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  },
  newEntry: {},
  history: { push: () => {} },
  location: { pathname: "/addEntry" }
};

export const entry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
