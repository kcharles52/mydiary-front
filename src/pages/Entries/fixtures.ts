export const props = {
  getEntriesAction: jest.fn(),
  deleteEntryAction: jest.fn(),
  entries: [
    {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      diaryTitle: "Title 2",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  deleted: [],
  handlesDelete: jest.fn(),
  history: {push:jest.fn()},
  location: { pathname: "/home" },
  match: { url: "/home" }
};

export const fetchedEntries =[
  {
    entry_id: 1,
    diaryTitle: "Title 1",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  },
  {
    entry_id: 2,
    diaryTitle: "Title 2",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  }
]