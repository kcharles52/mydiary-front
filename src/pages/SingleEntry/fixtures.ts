export const props = {
  getEntryAction: (id: string) => Promise.resolve(),
  deleteEntryAction: (id: string) => Promise.resolve(),
  fetchedEntry: { entry: {} },
  match: { params: { entry_id: "1" } },
  history: { push: jest.fn() },
  location: { pathname: "/Entry/1" }
};

export const fetchedEntry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
