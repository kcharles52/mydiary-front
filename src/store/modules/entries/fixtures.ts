export const intialState = {
  entries: [],
  entry: {},
  new: {},
  Message: "",
  deleted: []
};

export const fetchedEntries = {
  entries: [
    {
      entry_id: 1,
      title: "Title 1",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      title: "Title 2",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  Message:"Successfully fetched entries"
};

export const editedEntry = {
    entry_id: 1,
      title: "Edited",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
}