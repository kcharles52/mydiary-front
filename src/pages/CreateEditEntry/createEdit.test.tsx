//react libraries
import * as React from "react";

//third party libraries
import { mount, shallow } from "enzyme";

//fixtures
import { props, entry } from "./fixtures";

import { CreatEditEntry, mapDispatchToProps, mapStateToProps } from ".";

describe("Should render entries", () => {
  let wrapper, shallowWrapper;

  beforeEach(() => {
    wrapper = mount(<CreatEditEntry {...props} />);
    shallowWrapper = shallow(<CreatEditEntry {...props} />);
  });

  afterEach(() => {
    wrapper = shallowWrapper = null;
  });
  it("should render createEditForm ", () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it("should render edit form on Edit=true", () => {
    let props = {
      match: { params: { entry_id: "1" } },
      getEntryAction: Promise.resolve(entry),
      Edit: true,
      history: { push: jest.fn() }
    };
    wrapper.setProps(props);
    wrapper.setState({ Edit: true });
    wrapper.update();

    const spy = jest.spyOn(wrapper.instance(), "editDiaryEntry");
    wrapper.instance().forceUpdate();

    wrapper.find("button").simulate("click");
  });

  it("should create entry", () => {
    let props = {
      match: { params: { entry_id: undefined } },
      Edit: false,
      addEntryAction: jest.fn(),
      history: { push: jest.fn() }
    };

    const wrapper = mount(<CreatEditEntry {...props} />);
    const spy = jest.spyOn(wrapper.instance(), "createNewEntry");

    wrapper
      .find("input")
      .first()
      .simulate("change", {
        target: { value: "Test title", name: "diaryTitle" }
      });
    expect(wrapper.state().Entry.diaryTitle).toContain("Test title");

    wrapper.find("button").simulate("click");
    expect(spy).toBeCalled();
    spy.mockRestore();
  });
});

describe("Connect functions function", () => {
  describe("The mapStateToProps function", () => {
    it("should return the expected props object", () => {
      const state = {
        entries: { entry: entry, new: entry }
      };
      const props = mapStateToProps(state);
      expect(props.editEntry).toEqual(state.entries.entry);
      expect(props.newEntry).toEqual(state.entries.entry);
    });
  });

  describe("The mapDispatchToProps function", () => {
    let dispatch, props;
    beforeEach(() => {
      dispatch = jest.fn();
      props = mapDispatchToProps(dispatch);
    });
    it("should dispatch addEntry when addEntryAction is called", () => {
      props.addEntryAction();
      expect(dispatch).toHaveBeenCalled();
    });
    it("should dispatch editEntry when editEntryAction is called", () => {
      props.editEntryAction();
      expect(dispatch).toHaveBeenCalled();
    });
    it("should dispatch getEntry when getEntryAction is called", () => {
      props.getEntryAction();
      expect(dispatch).toHaveBeenCalled();
    });
  });
});
