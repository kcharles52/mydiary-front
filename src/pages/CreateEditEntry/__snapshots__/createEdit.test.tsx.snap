// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render entries should render createEditForm  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreatEditEntry
    Edit={false}
    addEntryAction={[MockFunction]}
    date="20-2-19"
    diaryBody="Test Diary body"
    diaryTitle="Test Diary Title"
    editEntry={
      Object {
        "entry": Object {
          "date": "12/12/2018",
          "delete": [MockFunction],
          "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
          "diaryTitle": "Title 1",
          "entry_id": 1,
        },
      }
    }
    editEntryAction={[MockFunction]}
    getEntryAction={
      [MockFunction] {
        "calls": Array [
          Array [
            "1",
          ],
          Array [
            "1",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
    history={
      Object {
        "push": [Function],
      }
    }
    location={
      Object {
        "pathname": "/addEntry",
      }
    }
    match={
      Object {
        "params": Object {
          "entry_id": "1",
        },
      }
    }
    newEntry={Object {}}
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
    title="Form title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <div
          className="CreateEntryForm"
        >
          <EntryForm
            Edit={false}
            addEntryAction={[MockFunction]}
            date="20-2-19"
            diaryBody="Test Diary body"
            diaryTitle="Test Diary Title"
            editEntry={
              Object {
                "entry": Object {
                  "date": "12/12/2018",
                  "delete": [MockFunction],
                  "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                  "diaryTitle": "Title 1",
                  "entry_id": 1,
                },
              }
            }
            editEntryAction={[MockFunction]}
            getEntryAction={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "1",
                  ],
                  Array [
                    "1",
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": Promise {},
                  },
                  Object {
                    "isThrow": false,
                    "value": Promise {},
                  },
                ],
              }
            }
            history={
              Object {
                "push": [Function],
              }
            }
            location={
              Object {
                "pathname": "/addEntry",
              }
            }
            match={
              Object {
                "params": Object {
                  "entry_id": "1",
                },
              }
            }
            newEntry={Object {}}
            onCancel={[Function]}
            onChange={[MockFunction]}
            onSubmit={[MockFunction]}
            title="Form title"
          />
        </div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <EntryForm
            Edit={false}
            addEntryAction={[MockFunction]}
            date="20-2-19"
            diaryBody="Test Diary body"
            diaryTitle="Test Diary Title"
            editEntry={
              Object {
                "entry": Object {
                  "date": "12/12/2018",
                  "delete": [MockFunction],
                  "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                  "diaryTitle": "Title 1",
                  "entry_id": 1,
                },
              }
            }
            editEntryAction={[MockFunction]}
            getEntryAction={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "1",
                  ],
                  Array [
                    "1",
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": Promise {},
                  },
                  Object {
                    "isThrow": false,
                    "value": Promise {},
                  },
                ],
              }
            }
            history={
              Object {
                "push": [Function],
              }
            }
            location={
              Object {
                "pathname": "/addEntry",
              }
            }
            match={
              Object {
                "params": Object {
                  "entry_id": "1",
                },
              }
            }
            newEntry={Object {}}
            onCancel={[Function]}
            onChange={[MockFunction]}
            onSubmit={[MockFunction]}
            title="Form title"
          />,
          "className": "CreateEntryForm",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "Edit": false,
            "addEntryAction": [MockFunction],
            "date": "20-2-19",
            "diaryBody": "Test Diary body",
            "diaryTitle": "Test Diary Title",
            "editEntry": Object {
              "entry": Object {
                "date": "12/12/2018",
                "delete": [MockFunction],
                "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                "diaryTitle": "Title 1",
                "entry_id": 1,
              },
            },
            "editEntryAction": [MockFunction],
            "getEntryAction": [MockFunction] {
              "calls": Array [
                Array [
                  "1",
                ],
                Array [
                  "1",
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": Promise {},
                },
                Object {
                  "isThrow": false,
                  "value": Promise {},
                },
              ],
            },
            "history": Object {
              "push": [Function],
            },
            "location": Object {
              "pathname": "/addEntry",
            },
            "match": Object {
              "params": Object {
                "entry_id": "1",
              },
            },
            "newEntry": Object {},
            "onCancel": [Function],
            "onChange": [MockFunction],
            "onSubmit": [MockFunction],
            "title": "Form title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <div
            className="CreateEntryForm"
          >
            <EntryForm
              Edit={false}
              addEntryAction={[MockFunction]}
              date="20-2-19"
              diaryBody="Test Diary body"
              diaryTitle="Test Diary Title"
              editEntry={
                Object {
                  "entry": Object {
                    "date": "12/12/2018",
                    "delete": [MockFunction],
                    "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                    "diaryTitle": "Title 1",
                    "entry_id": 1,
                  },
                }
              }
              editEntryAction={[MockFunction]}
              getEntryAction={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "1",
                    ],
                    Array [
                      "1",
                    ],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": Promise {},
                    },
                    Object {
                      "isThrow": false,
                      "value": Promise {},
                    },
                  ],
                }
              }
              history={
                Object {
                  "push": [Function],
                }
              }
              location={
                Object {
                  "pathname": "/addEntry",
                }
              }
              match={
                Object {
                  "params": Object {
                    "entry_id": "1",
                  },
                }
              }
              newEntry={Object {}}
              onCancel={[Function]}
              onChange={[MockFunction]}
              onSubmit={[MockFunction]}
              title="Form title"
            />
          </div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <EntryForm
              Edit={false}
              addEntryAction={[MockFunction]}
              date="20-2-19"
              diaryBody="Test Diary body"
              diaryTitle="Test Diary Title"
              editEntry={
                Object {
                  "entry": Object {
                    "date": "12/12/2018",
                    "delete": [MockFunction],
                    "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                    "diaryTitle": "Title 1",
                    "entry_id": 1,
                  },
                }
              }
              editEntryAction={[MockFunction]}
              getEntryAction={
                [MockFunction] {
                  "calls": Array [
                    Array [
                      "1",
                    ],
                    Array [
                      "1",
                    ],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": Promise {},
                    },
                    Object {
                      "isThrow": false,
                      "value": Promise {},
                    },
                  ],
                }
              }
              history={
                Object {
                  "push": [Function],
                }
              }
              location={
                Object {
                  "pathname": "/addEntry",
                }
              }
              match={
                Object {
                  "params": Object {
                    "entry_id": "1",
                  },
                }
              }
              newEntry={Object {}}
              onCancel={[Function]}
              onChange={[MockFunction]}
              onSubmit={[MockFunction]}
              title="Form title"
            />,
            "className": "CreateEntryForm",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "Edit": false,
              "addEntryAction": [MockFunction],
              "date": "20-2-19",
              "diaryBody": "Test Diary body",
              "diaryTitle": "Test Diary Title",
              "editEntry": Object {
                "entry": Object {
                  "date": "12/12/2018",
                  "delete": [MockFunction],
                  "diaryEntryBody": "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
                  "diaryTitle": "Title 1",
                  "entry_id": 1,
                },
              },
              "editEntryAction": [MockFunction],
              "getEntryAction": [MockFunction] {
                "calls": Array [
                  Array [
                    "1",
                  ],
                  Array [
                    "1",
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": Promise {},
                  },
                  Object {
                    "isThrow": false,
                    "value": Promise {},
                  },
                ],
              },
              "history": Object {
                "push": [Function],
              },
              "location": Object {
                "pathname": "/addEntry",
              },
              "match": Object {
                "params": Object {
                  "entry_id": "1",
                },
              },
              "newEntry": Object {},
              "onCancel": [Function],
              "onChange": [MockFunction],
              "onSubmit": [MockFunction],
              "title": "Form title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
